examples_tests/43A.SumAndCDFFilters/main.cpp

```diff
@@ -56,20 +58,65 @@ int main()
 			newImageParams.format = EF_R32G32B32A32_SFLOAT;
 			#endif // IMAGE_VIEW
 
-			auto newRegions = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<ICPUImage::SBufferCopy>>(referenceRegions.size());
+			auto newRegions = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<ICPUImage::SBufferCopy>>
+			(
+				#ifdef IMAGE_VIEW
+				referenceRegions.size()
+				#else
+				2u
+				#endif // IMAGE_VIEW
+			);
+
 			size_t regionOffsets = {};
 
+			#ifdef IMAGE_VIEW
 			for (auto newRegion = newRegions->begin(); newRegion != newRegions->end(); ++newRegion)
 			{
+				/*
+					Regions pulled directly from a loader doesn't overlap, so each following is a certain single mipmap
+				*/
+
+				auto idOffset = newRegion - newRegions->begin();
 				*newRegion = *(referenceRegion++);
 				newRegion->bufferOffset = regionOffsets;
-				regionOffsets += newRegion->imageExtent.width * newRegion->imageExtent.height * newRegion->imageExtent.depth * newRegion->imageSubresource.layerCount * asset::getTexelOrBlockBytesize(newImageParams.format);
+
+				const auto fullMipMapExtent = image->getMipSize(idOffset);
+
+				regionOffsets += fullMipMapExtent.x * fullMipMapExtent.y * fullMipMapExtent.z * newImageParams.arrayLayers * asset::getTexelOrBlockBytesize(newImageParams.format);
 			}
+			#else
+
+			/*
+				2 overlapping regions
+			*/
```

Comment:
erm IMO validation should return false and not let setting multiple regions defining same image pixels. Why would it ever be correct/useful? @devshgraphicsprogramming ?

Reply:
Because Vulkan allows it and its well defined.